### PR TITLE
makima: init at 0.4.3

### PR DESCRIPTION
**Pull request by ByteSudoer**
## Description of changes

Makima is a daemon for Linux to remap keyboards, mice, controllers and tablets.
homepage URL: N/A
source URL: https://github.com/cyber-sushi/makima
license: GPLv3
platforms: linux

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [X] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by keenanweaver** [comment]
Result of `nixpkgs-review pr 297604` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>makima</li>
  </ul>
</details>

**Review by keenanweaver — APPROVED** [review]
Works great. Thank you very much!

File: pkgs/by-name/ma/makima/package.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  buildInputs = [ udev ];
+
+  meta = with lib; {
+    description = "Linux daemon to remap and create macros for keyboards, mice and controllers ";
+    homepage = "https://github.com/cyber-sushi/makima";
+    license = with licenses; [ gpl3 ];
+    maintainers = with maintainers; [ ByteSudoer ];
+    mainProgram = "makima";
+  };
+
```
Comment (Aleksanaa) [inline pkgs/by-name/ma/makima/package.nix:31]:
```suggestion
```

File: pkgs/by-name/ma/makima/package.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  cargoHash = "sha256-CWqKV9MKnUSo9d5insrS1san7EtQvB3h5nY05ey5ZvQ=";
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ udev ];
+
+  meta = with lib; {
+    description = "Linux daemon to remap and create macros for keyboards, mice and controllers ";
+    homepage = "https://github.com/cyber-sushi/makima";
+    license = with licenses; [ gpl3 ];
```
Comment (Aleksanaa) [inline pkgs/by-name/ma/makima/package.nix:27]:
```suggestion
    license = licenses.gpl3Plus;
```

Don't use `licenses.gpl3`

File: pkgs/by-name/ma/makima/package.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    hash = "sha256-hHTWZ8FY+r+X7LTp+GvO9m58u8/zsaxmMJMNMb0KRaU=";
+  };
+
+  cargoHash = "sha256-CWqKV9MKnUSo9d5insrS1san7EtQvB3h5nY05ey5ZvQ=";
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ udev ];
+
+  meta = with lib; {
+    description = "Linux daemon to remap and create macros for keyboards, mice and controllers ";
```
Comment (Aleksanaa) [inline pkgs/by-name/ma/makima/package.nix:25]:
```suggestion
    description = "Linux daemon to remap and create macros for keyboards, mice and controllers";
```

File: pkgs/by-name/ma/makima/package.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ udev ];
+
+  meta = with lib; {
+    description = "Linux daemon to remap and create macros for keyboards, mice and controllers";
+    homepage = "https://github.com/cyber-sushi/makima";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ ByteSudoer ];
+    mainProgram = "makima";
```
Comment (Aleksanaa) [inline pkgs/by-name/ma/makima/package.nix:29]:
```suggestion
    mainProgram = "makima";
    platforms = platforms.linux;
```

Since it's targeting linux

**Comment by keenanweaver** [comment]
v0.4.3 has been released: https://github.com/cyber-sushi/makima/releases/tag/v0.4.3

**Comment by ByteSudoer** [comment]
I will update the derivation accordingly. It hasn't been merged yet anyway . Thanks for the heads up.

**Comment by keenanweaver** [comment]
Result of `nixpkgs-review pr 297604` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>makima</li>
  </ul>
</details>

**Review by keenanweaver — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**